tivars/var.py

```diff
@@ -111,137 +61,163 @@ def checksum(self) -> int:
+    def __init__(self, *, name: str = 'UNNAMED', model: 'TIModel' = None):
+        signature = "**TI83F*" if model is None else model.signature
 
-        file.read(51)
+        meta_length = TIEntry.flash_meta_length if \
+            model is None or model.has(TIFeature.FLASH) else TIEntry.base_meta_length
 
-        type_id = file.read(1)
-        try:
-            var = TIVar._type_ids[type_id](file.name, model)
-        except KeyError:
-            raise ValueError("File has unknown type ID.")
+        super().__init__(signature=signature)
+        super(TIHeader, self).__init__(meta_length=meta_length,
+                                       name=name)
+        self.model = model
+
+    @property
+    def entry(self) -> 'TIEntry':
+        return TIEntry(self.meta_length, self.name, self.version, self.archived, self.data)
+
+    @property
+    def extension(self) -> str:
+        return self.extensions[self.model]
 
-        file.seek(0)
-        var.load(file)
-        return var
+    @property
+    def header(self) -> 'TIHeader':
+        return TIHeader(self.signature, self.export, self.comment, self.entry_length)
 
     @staticmethod
     def register(var_type: type['TIVar']):
-        TIVar._type_ids[var_type.type_id] = var_type
+        TIVar.type_ids[var_type.type_id] = var_type
 
     def archive(self):
-        if self.model.has(TIFeature.FLASH):
+        if self.model is None or self.model.has(TIFeature.FLASH):
             self.archived = True
         else:
-            raise TypeError("Calculator model does not support archiving.")
+            raise TypeError(f"The {self.model} does not support archiving.")
+
+    def bytes(self) -> bytes:
+        dump = b''
+
+        dump += self.signature_bytes
+        dump += self.export
+        dump += self.comment_bytes
+        dump += self.entry_length_bytes
+
+        dump += self.meta_length_bytes
+        dump += self.data_length_bytes
+        dump += self.type_id
+        dump += self.name_bytes
+
+        if self.flash_bytes:
+            dump += self.version
+            dump += self.archived_bytes
+
+        dump += self.data_length_bytes
+        dump += self.data
+        dump += self.checksum_bytes
 
-    def dump(self) -> bytes:
-        return self.header.dump() + self.entry.dump() + self.checksum.to_bytes(2, 'little')
+        return dump
 
-    def dumps(self) -> str:
-        pass
+    def load_bytes(self, data: bytes | Iterator[bytes]):
+        data = iter(data)
 
-    def load(self, file: BinaryIO):
-        self.signature = file.read(8)
-        self.export = file.read(3)
-        self.comment = file.read(42).decode('utf8')
-        entry_length = int.from_bytes(file.read(2), 'little')
+        self.load_buffer(TIVar.signature, data)
+        match self.signature:
+            case TI_82.signature:
+                model = TI_82
+            case TI_83.signature:
+                model = TI_83
+            case TI_84p.signature:
+                model = TI_84p
```

Comment:
Wouldn't it be less failure-prone to assume the model with the most feature bits; i.e. the python CE?

Reply:
Probably